scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -651,6 +651,13 @@ def run_cmake(self, args="", filter_stages=None):
             f'-DPython3_EXECUTABLE={pathlib.Path(sys.executable).as_posix()}'
         ]
 
+        if self.instance.testsuite.harness == 'bsim':
+            cmake_args.extend([
+                '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON',
+                '-DCONFIG_ASSERT=y',
```

Comment:
If you enable this by default, should we remove the `CONFIG_ASSERT=y` from the config files of the affected tests?

Reply:
We can, but it's always been enabled in compile.source, but still present in conf files, probably just copy-pasted over from one to another, or may be it was added to compile source later.